Remove unnecessary noCuda arg from AtomicJitFuture

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56984 Use a FutureFactoryRegistry to allow libtorch_cpu files to create CUDAFuture
* **#56973 Remove unnecessary noCuda arg from AtomicJitFuture**



Differential Revision: [D28020918](https://our.internmc.facebook.com/intern/diff/D28020918)

mrshenli committed:
Remove unnecessary noCuda arg from AtomicJitFuture

[ghstack-poisoned]

lw reviewed (approved):
Brilliant! I was thinking the same yesterday when looking at this, I'm glad to see it fixed!

mrshenli committed:
Update on "Remove unnecessary noCuda arg from AtomicJitFuture"




Differential Revision: [D28020918](https://our.internmc.facebook.com/intern/diff/D28020918)

[ghstack-poisoned]
